src/aead/less_safe_key.rs

```diff
@@ -94,28 +92,18 @@ impl LessSafeKey {
     where
         A: AsRef<[u8]>,
     {
-        let ciphertext_and_tag_len = in_out
+        // XXX: There are redundant dynamic checks here. Do they get optimized away?
+        let tag_offset = in_out
             .len()
-            .checked_sub(ciphertext_and_tag.start)
-            .ok_or(error::Unspecified)?;
-
-        let ciphertext_len = ciphertext_and_tag_len
             .checked_sub(TAG_LEN)
             .ok_or(error::Unspecified)?;
 
-        let (in_out, received_tag) = in_out.split_at_mut(ciphertext_and_tag.start + ciphertext_len);
-
+        // Split the tag off the end of `in_out`.
+        let (in_out, received_tag) = in_out.split_at_mut(tag_offset);
         let received_tag = (*received_tag).try_into()?;
+        let ciphertext: RangeFrom<_> = ciphertext_and_tag;
 
-        open_within_(
-            self,
-            nonce,
-            Aad::from(aad.as_ref()),
-            received_tag,
-            in_out,
-            ciphertext_and_tag,
-            ciphertext_len,
-        )
+        self.open_in_place_separate_tag(nonce, aad, received_tag, in_out, ciphertext)
```

Comment:
This makes it clear that `open_in_place_separate_tag` is the most fundamental (public) operation.